src/lib/prov/tpm2/tpm2_error.h

```diff
@@ -0,0 +1,42 @@
+/*
+* TPM 2 error handling
+* (C) 2024 Jack Lloyd
+* (C) 2024 René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity GmbH
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_TPM2_ERROR_H_
+#define BOTAN_TPM2_ERROR_H_
+
+#include <botan/exceptn.h>
+
+/// Forward declaration of TSS2 type for convenience
+using TSS2_RC = uint32_t;
```

Comment:
These forward declares and "forward typedefs" are a bit dirty anyway, as they will cause trouble if the TSS would ever change their underlying types (obviously). I'm fairly confident that that won't do that, given that this API is standardized and agreed upon by the TCG.

We chose to put them into global namespace, because they _are_ (or will be) in global namespace in the TSS headers as well. Moving them into `Botan::TPM2` would be possible, I think. No strong opinion on that, to be honest.

That said: Perhaps it would be a good idea to move all those forward declares into a dedicated header (say: `tpm2_forward_declares.h`).